conventional-changelog-lighthouse/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+const readFileSync = require('fs').readFileSync;
+const resolve = require('path').resolve;
+const mainTemplate = readFileSync(resolve(__dirname, 'templates/template.hbs')).toString();
+const headerPartial = readFileSync(resolve(__dirname, 'templates/header.hbs')).toString();
+const commitPartial = readFileSync(resolve(__dirname, 'templates/commit.hbs')).toString();
+
+const parserOpts = {
+  headerPattern: /^(\w*)(?:\((.*)\))?: (.*)$/,
+  headerCorrespondence: [
+    'type',
+    'scope',
+    'message',
+  ],
+};
+
+const writerOpts = {
+  mainTemplate,
+  headerPartial,
+  commitPartial,
+  transform: commit => {
+    if (typeof commit.hash === 'string') {
+      commit.hash = commit.hash.substring(0, 7);
+    }
+
+    if (commit.type === 'test') {
+      commit.type = 'tests';
+    }
+
+    if (commit.type) {
+      commit.type = commit.type.replace(/_/g, ' ');
+      commit.type = commit.type.substring(0, 1).toUpperCase() + commit.type.substring(1);
+    } else {
+      commit.type = 'Misc';
+    }
+
+    let pullRequestMatch = commit.header.match(/\(#(\d+)\)/);
+    // if header does not provide a PR we try the message
+    if (!pullRequestMatch && commit.message) {
+      pullRequestMatch = commit.message.match(/\(#(\d+)\)/);
+    }
+
+    if (pullRequestMatch) {
+      commit.header = commit.header.replace(pullRequestMatch[0], '');
+      if (commit.message) {
+        commit.message = commit.message.replace(pullRequestMatch[0], '');
+      }
+
+      commit.PR = pullRequestMatch[1];
+    }
+
+    return commit;
+  },
+  groupBy: 'type',
+  commitGroupsSort: (a, b) => {
+    // put new audit on the top
+    if (b.title === 'New audit') {
+      return 1;
+    }
+
+    return a.title.localeCompare(b.title);
+  },
+  commitsSort: ['type', 'scope', 'message'],
```

Comment:
In the past we've only sorted by `type` and `scope` (if it exists) and then by commit order, but not sure if that's possible with this